### PR TITLE
Update samples readme to match current  projects

### LINKED ISSUE #757: samples/readme.md needs to reflect current state
The readme.md in the /samples directory is months old, and does not represent the current state of the .NET Core product, or the samples in the folder. 

It should represent the current contents, or be forward looking for where we want to go

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #757

Update this readme to reflect that all samples are under this heading. Also, provide guidance on where samples should go, and how to link from the sample to the article and vice-versa.

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
+are organized in sub-folders. These sub-folders are organized similar
+to the organization of the docs for .NET Core.
 
-Sample projects are broken up by language.  Here is an example general structure:
+Some of the articles will have more than one sample associated with them. 
+
+The readme.md file for each sample will refer to the article so that
+you can read more about the concepts covered in each sample.
+
+## building a sample
```
Comment (svick) [inline samples/README.md]: Capitalization: building → Building

File: samples/README.md [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
-            project.json
-      /vb
-      /fsharp
+dotnet restore
+dotnet build
+dotnet run
 ```
+
+These will install any needed dependencies, build the project, and run
+the project respetively.
```
Comment (svick) [inline samples/README.md]: Typo: respetively

File: samples/README.md [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+
+## Creating new samples
+
 If you wish to add a code sample:
 
 1. Your sample **must be part of a buildable project**
 2. Your sample **cannot be a Visual Studio Project**
-	- We do not want Windows and Visual Studio to be a dependency for people building these on their own.
+	- We do not want Windows and Visual Studio to be a dependency for people building these on their own. The only exception is if your sample highlights particular tooling and is referenced by a topic that explains the pre-requisites.
 3. Your sample shoud conform to the [corefx coding style](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) to maintain consistency.
```
Comment (mairaw) [inline samples/README.md]: typo: shoud

File: samples/README.md [hunk 4]
```diff
@@ -66,18 +66,21 @@ add a run command specifying the sample folder so people using VSCode or Visual
   ```
   To to build and run your sample...
```
Comment (mairaw) [inline samples/README.md]: Double to's

File: samples/README.md [hunk 5]
```diff
@@ -66,18 +66,21 @@ add a run command specifying the sample folder so people using VSCode or Visual
...
+    dotnet build
  ```
 7. Run!
 
  ```
-    dnx run
+    dotnet run
  ```
+
+8. Add a readme.md
```
Comment (mairaw) [inline samples/README.md]: Should this say "Add a readme.md file in the sample folder"?

File: samples/README.md [hunk 6]
```diff
@@ -1,28 +1,46 @@
...
+
+These will install any needed dependencies, build the project, and run
+the project respetively.
+
+Multi-project samples have instructions in their root directory in
+a `README.md` file.  
+
+Except where noted, all samples will build from the command line, on
+any platform supported by .NET Core. There are a few samples that are
+specific to Visual Studio, and will require Visual Studio 2015, Update 3
```
Comment (mairaw) [inline samples/README.md]: nit: no need for comma separating 2015 and Update 3

**Comment by mairaw** [comment]
LGTM. Left a couple of comments (mostly for existing issues we didn't catch before)

File: samples/README.md [hunk 7]
```diff
@@ -1,28 +1,46 @@
...
+and will require a specific platform.
+
+## Creating new samples
+
 If you wish to add a code sample:
 
 1. Your sample **must be part of a buildable project**
 2. Your sample **cannot be a Visual Studio Project**
-	- We do not want Windows and Visual Studio to be a dependency for people building these on their own.
+	- We do not want Windows and Visual Studio to be a dependency for people building these on their own. The only exception is if your sample highlights particular tooling and is referenced by a topic that explains the pre-requisites.
```
Comment (rpetrusha) [inline samples/README.md]: typo: prerequisite

**Comment by BillWagner** [comment]
Responded to all feedback, and merging.

**Pull request merged**